src/stringProcessing/url.js

```diff
@@ -81,6 +81,21 @@ function getHostname( url ) {
 	return url.hostname;
 }
 
+/**
+ * Returns the protocol of a URL.
+ *
+ * @param {string} url The URL to retrieve the protocol of.
+ * @returns {string|null} The protocol of the URL.
+ */
+function getProtocol( url ) {
+	const protocolMatcher = /^(?:[a-z]+:)?\/\//i;
+	const protocolMatch = protocolMatcher.exec( url );
+	if ( ! protocolMatch ) {
+		return null;
+	}
```

Comment:
this translates to "if null, then return null": works but maybe coud be a bit more clear?